plutus-core/plutus-ir/test/TransformSpec.hs

```diff
@@ -222,6 +223,39 @@ inline =
             , "letNonPureMultiStrict"
             ]
 
+-- | Tests to ensure global uniqueness.
```

Comment:
I don't think this is a good way to do this. It's adding a huge amount of duplicated tests that we have to check by eye. Why not add a programmatic check for global uniqueness to the existing tests? I don't think we have a function for that directly, but we could write one.

Such a check would benefit us in every other place where we want to be sure we have global uniqueness, for example.

(or is there a faster way? @effectfully ?)

Reply:
> Why not add a programmatic check for global uniqueness to the existing tests? I don't think we have a function for that directly, but we could write one.

I thought it was `PlutusCore.Check.Uniques.checkProgram` that you implemented? The one used in `[propRenameCheck](https://github.com/input-output-hk/plutus/blob/e9cbc3e84c0f17967cd92d9d26321ffd8ec08e23/plutus-core/plutus-core/test/Check/Spec.hs#L26)` referenced by Marty. I'm very confused otherwise!

Reply:
Aha, yes, it does look like it will do that, I had forgotten.

Reply:
Yeah. Thanks Roman. I'm wondering if we can trust that. It's not such a simple property that we can be sure it does the right thing it seems? What do you two think? Either way I feel unit tests can't hurt though. I was going to set up nicer unit tests today.

Reply:
> I'm wondering if we can trust that. It's not such a simple property that we can be sure it does the right thing it seems? What do you two think?

It was Michael who implemented the uniqueness check, so I'm not quite sure, but I'd expect 

```haskell
Uniques.checkProgram (\case { FreeVariable{} -> False; IncoherentUsage {} -> False; _ -> True})
```

taken from `propRenameCheck` to work just as well in your case? I don't think you can reuse `propRenameCheck` directly, if that's what you're asking, but you should be able to reuse the line from it that I just quoted.

Well except maybe in your case you also want to throw an error upon encountering a free variable, I don't know, haven't looked into the inliner yet (and won't do it until much later unfortunately).

Reply:
Thanks. As discussed I'll make a PIR version of `checkProgram` and add it to the existing `inline` tests.